modules/phase_field/include/userobjects/ComputeExternalGrainForceAndTorque.h

```diff
@@ -0,0 +1,77 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef COMPUTEEXTERNALGRAINFORCEANDTORQUE_H
+#define COMPUTEEXTERNALGRAINFORCEANDTORQUE_H
+
+#include "ShapeElementUserObject.h"
+#include "GrainForceAndTorqueInterface.h"
+#include "DerivativeMaterialPropertyNameInterface.h"
+
+//Forward Declarations
+class ComputeExternalGrainForceAndTorque;
+class GrainTrackerInterface;
+
+template<>
+InputParameters validParams<ComputeExternalGrainForceAndTorque>();
+
+/**
+ * This class is here to get the force and torque acting on a grain
+ */
+class ComputeExternalGrainForceAndTorque :
+    public ShapeElementUserObject,
+    public DerivativeMaterialPropertyNameInterface,
```

Comment:
I recommend inheriting from `DerivativeMaterialInterface<ShapeElementUserObject>` instead. Otherwise you do not get the optional material property capability.